tests/unit/models/gpflow/test_sampler.py

```diff
@@ -719,3 +723,48 @@ def test_rff_and_decoupled_trajectory_give_similar_results(noise_var: float) ->
     npt.assert_allclose(
         tf.math.reduce_variance(eval_1, 1), tf.math.reduce_variance(eval_2, 1), rtol=1.0
     )  # variance across samples should (very) roughly agree for different samplers
+
+
+@pytest.mark.parametrize(
+    ("batch_shape", "n_sample_dim"),
+    [
+        ([5], 1),
+        ([5], 2),
+        ([5, 5], 2),
+        ([1, 2, 3, 4], 5),
+    ],
+)
+def test_idealised_normal_samples__various_shapes(
+    batch_shape: list[int], n_sample_dim: int
+) -> None:
+    samples = idealised_normal_samples(tf.TensorShape(batch_shape), n_sample_dim)
+    assert samples.shape == batch_shape + [
+        n_sample_dim,
+    ]
+
+
+def test_kl_divergence_normal_samples() -> None:
```

Comment:
change name to `test_idealised_samples_return_standard_normal_samples` ? (took me a little while to get what this was testing)